nservicebus/messaging/evolving-contracts.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Evolving messages contracts
+reviewed: 2016-11-08
+component: Core
+---
+
+In message-based systems the messages are part of the contract, which defines how services communicate with each other.
+
+Evolving contracts over time is not an easy task and the appropriate strategy should be defined individually for each system. Generally that problem can't be resolved on the infrastructure level, therefore NServiceBus users need to analyze their systems, consider how they expect them to evolve and define the strategy which will make the most sense in their particular circumstances.
+
+This article presents basic guidelines for choosing contracts evolution strategy, avoiding common mistakes and ensuring the contracts will be easy to evolve over time.
+
+
+## Designing contracts
+
+
+### Messages
+
+Messages should ideally be:
```

Comment:
shouldnt this advice exist on this page https://docs.particular.net/nservicebus/messaging/messages-events-commands and linked to from here ?